### PR TITLE
Use date dialog

### PR DESCRIPTION
**Pull request by bethanclarke**
Changes have been made to:
- the order of columns in the design 
- pentad function 
- split date function - shifting my month implemented, and numeric and abbr for pentad and dekad

### CODE REVIEW

**Comment by bethanclarke** [comment]
This work corresponds to issue #4603 
Still to do is abbr quarter

**Comment by bethanclarke** [comment]
@rdstern and @Patowhiz this is ready for testing

**Comment by Patowhiz** [comment]
Testing this

**Comment by Patowhiz** [comment]
@bethanclarke I'm getting an error when I check Pentad Abbrv. check the screenshot below

use date error

**Comment by Patowhiz** [comment]
On the same, the Pentad Abbrv check is enabled, but when I check it only, the button ok is not enabled. I'm suspecting the event that calls on test ok is not being called

**Comment by Patowhiz** [comment]
Dekad Abbrv does not enable the button ok. Probably the same issue mentioned above. (Test Ok not called by its event value changed).

**Comment by Patowhiz** [comment]
@rdstern @dannyparsons @bethanclarke with the new position of starting month, would it not give the impression that all the options are shiftable? or it's just common sense?

impress

File: instat/DlgUseDate.vb [hunk 1]
```diff
@@ -208,7 +208,7 @@ Public Class dlgUseDate
...
+    Private Sub ucrControls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverUseDate.ControlContentsChanged, ucrChkWeekName.ControlContentsChanged, ucrChkWeekdayNum.ControlContentsChanged, ucrChkWeekNum.ControlContentsChanged, ucrChkShiftPentadNum.ControlContentsChanged, ucrChkShiftPentadAbbr.ControlContentsChanged, ucrChkShiftDekadAbbr.ControlContentsChanged, ucrChkWeekAbbr.ControlContentsChanged, ucrChkShiftMonthNum.ControlContentsChanged, ucrChkLeapYearNum.ControlContentsChanged, ucrChkWeekdayName.ControlContentsChanged, ucrChkShiftMonthName.ControlContentsChanged, ucrChkShiftDekadNum.ControlContentsChanged, ucrChkDayInMonthNum.ControlContentsChanged, ucrChkDayInYearNum.ControlContentsChanged, ucrChkWeekdayAbbr.ControlContentsChanged, ucrChkShiftMonthAbbr.ControlContentsChanged, ucrChkShiftDayInYearNum366.ControlContentsChanged, ucrChkShiftYearNum.ControlContentsChanged, ucrChkShiftDayInYearNum366.ControlContentsChanged, ucrChkShiftQuarterNum.ControlContentsChanged, ucrInputComboBoxStartingMonth.ControlContentsChanged
```
Comment (Patowhiz) [inline instat/DlgUseDate.vb]: looks good

Comment (Patowhiz) [reply instat/DlgUseDate.vb]: @bethanclarke have you checked the error?

Comment (bethanclarke) [reply instat/DlgUseDate.vb]: What is the error occurring here?

**Comment by Patowhiz** [comment]
@bethanclarke I'm also getting an error when I try shifting Day in Year 366. Probably a slight logical error in the R Function.

error2

File: instat/DlgUseDate.vb (hunk 1 above)
Comment (Patowhiz) [reply instat/DlgUseDate.vb]: Sorry, I was referring to the errors I have mention on the pull request conversation thread.

**Comment by bethanclarke** [comment]
@dannyparsons i still have the problem with shifting Day in Year 366.  I cannot figure out what is going wrong with s_doy, could you please take a look? Thanks

**Comment by bethanclarke** [comment]
@dannyparsons the error is in line 2142:
if(self$is_climatic_data()) self$set_climatic_types(types = c(s_doy = col_name))
when this is commented out day in year (366) works when shifting by month

**Comment by maxwellfundi** [comment]
@Patowhiz Have you looked at this again?

**Comment by dannyparsons** [comment]
The climatic type needs to be added to the list of climatic types for this to work. If you go to `data_object_R6.R` around line 2189 is where the climatic types are defined so you need to add something like: `s_doy_label="s_doy"` and then add that label to the list on line 2223. Then it is a recognised climatic type and this should work.

> @dannyparsons the error is in line 2142:
> if(self$is_climatic_data()) self$set_climatic_types(types = c(s_doy = col_name))
> when this is commented out day in year (366) works when shifting by month

**Comment by Muthenya** [comment]
I thought the starting month was to move below the first table.

image

**Comment by bethanclarke** [comment]
@dannyparsons thank you Danny, I will add that to climatic type. 

I think having starting month above the first table with group boxes is ok - I think it is clear that starting month only applies to the first table. But I can change this if people think otherwise.

**Comment by bethanclarke** [comment]
@dannyparsons included s_doy as climatic and now there is no issues

**Comment by Muthenya** [comment]
This looks good.

**Comment by maxwellfundi** [comment]
@dannyparsons  this is ready for you

**Pull request merged**